src/rascal/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -633,10 +633,10 @@ namespace rascal {
     }
 
     //! Returns atom type given an atom tag, also works for ghost atoms
-    int & get_atom_type(int atom_tag) {
-      // return this->atom_types[this->get_atom_index(atom_tag)];
-      return this->atom_types[atom_tag];
-    }
+    int & get_atom_type(int atom_tag) { return this->atom_types[atom_tag]; }
```

Comment:
Why is this function returning a non-const reference? Are we writing to it anywhere?

Reply:
I am not quite sure. That's just how it was.

Reply:
Can you remove it and only provide the other one then?